### PR TITLE
org.freedesktop.Sdk.Extension.dart

### PR DESCRIPTION
**Pull request by carterbox**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] ~~I am an upstream contributor to the project.~~ If not, I contacted upstream developers about submitting their software to Flathub. **Link: https://github.com/dart-lang/sdk/issues/50881**
- [ ] ~~I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).~~ I do not own the domain. This is a Freedesktop SDK extension.
- [x] ~~Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*~~ No patches.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

AFAIK, you cannot submit multiple flatpaks in one PR. I am opening this PR because in https://github.com/flathub/flathub/pull/3758 I was encouraged to create SDK extensions for dart and flutter.

I based this PR on the flathub recipes for the golang and rust SDK extensions. One thing that I didn't see/understand was how to test this extension. I added a `test-commands` section to a module, but I didn't see it executed when I built the recipe locally.

### CODE REVIEW

File: org.freedesktop.Sdk.Extension.dart.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      - cp -rpv * /usr/lib/sdk/dart/
+  - name: scripts
+    modules:
+      - dart
+    sources:
+      - type: script
+        commands:
+          - export DARTROOT=/usr/lib/sdk/dart
+          - export PATH=$PATH:$DARTROOT/bin
+        dest-filename: enable.sh
```
Comment (carterbox) [inline org.freedesktop.Sdk.Extension.dart.yaml:44]: I'm not sure that this works as intended. I tried sourcing this enable script locally and it didn't add dart to the path.

**Comment by barthalion** [comment]
bot, build org.freedesktop.Sdk.Extension.dart

**Comment by carterbox** [comment]
bot, build org.freedesktop.Sdk.Extension.dart

File: org.freedesktop.Sdk.Extension.dart.yaml (hunk 1 above)
Comment (carterbox) [reply org.freedesktop.Sdk.Extension.dart.yaml]: NVM. The extension is loaded automatically into the environment if it is installed. This wasn't clear to me as a new flatpak user.

**Comment by carterbox** [comment]
Because Flathub/Flatpak disallows internet access during the build step, it doesn't make sense to support Dart unless they implement a vendoring feature which makes it easy to package dependencies along with project source.

**Pull request closed without merging**